### ISSUE TITLE
Improve event date / time display

### ISSUE THREAD

**Issue by ztNFny** [cgeo/cgeo#16303]
GC recently removed the display of event times from listings (https://forums.geocaching.com/GC/index.php?/topic/410241-release-notes-website-and-geocaching%C2%AE-app-event-start-and-end-times-october-17-2024/) and judging by the latest comment it seems that info is not going to return.

We should use this as an opportunity to consider whether we want to display the event information in a "better" way.

Currently the date and time is prepended to the listing text:
![image](https://github.com/user-attachments/assets/3155f89e-cb15-4dc3-b007-561d01f88003)
Advantages for user:
- high visibility (imho top of the listing is THE place for the info as it's THE main info about an event)
Disadvantages for users:
- unstructured information (?)

Advantages for programmer:
- all the existing code using event times (e.g. "add to calendar") continues working as is
- no need for additional database fields
Disdvantages for programmer:
- unstructured information
- to consume the event times somewhere (e.g. "add to calendar") the listing text has to be parsed

Alternatives:
- Top of listing, but structured (e.g. "Date: xxx \n Time: 00:00 - 01:00")
- CacheDetails as "metadata" field, next to the Date field
  - (imho not good as it splits the event info then onto 2 tabs)
- other ideas?

PS: This is relevant for GC events only. On OC the event time information is not contained in the metadata at all.

**Comment by SammysHP** [comment]
See also #16229.

**Comment by baiti** [comment]
Is this one really still open?

**Comment by ztNFny** [comment]
> Is this one really still open?

Yes - it's still displayed in that unstructured way i implemented ~1.5 years ago 😉

**Comment by baiti** [comment]
> We should use this as an opportunity to consider whether we want to display the event information in a "better" way. 

Do you have an idea how? Is it also about adding the details to database fields? (For programmatic exploitation elsewhere, as you rightfully stated)

**Comment by moving-bits** [comment]
Adding time info to the database probably is the way to go. This would, among other things, allow for additionally displaying the time in cache infosheet, which would be a huge advantage when browsing through events on the map.

**Comment by ztNFny** [comment]
> This would, among other things, allow for additionally displaying the time in cache infosheet

Only for caches already downloaded - and for those it should already be possible.by parsing the listing on displaying the popup.
